beacon-chain/blockchain/execution_engine_test.go

```diff
@@ -191,6 +196,49 @@ func Test_NotifyForkchoiceUpdate(t *testing.T) {
 	}
 }
 
+func Test_NotifyForkchoiceUpdateRecursive(t *testing.T) {
```

Comment:
Can we add more useful checks that will actually test for the behavior we want? For example, we want to ensure that forkchoice is updated with a correct head after fork choice selection upon first receiving an invalid payload from EL. If we can simulate a fork where one branch contains an invalid payload and another is valid, we should prove that forkchoice was updated to the better head, and that forkchoice updated was called with the right information